ml_tools/eolearn/ml_tools/TDigestTask.py

```diff
@@ -0,0 +1,178 @@
+        compute quantiles of the underlying dataset into eo-learn.
+    The output features of the tasks may be merged to compute a representation of the complete dataset.
+    That enables quantile based normalisation or statistical analysis of datasets larger than RAM in EO.
+    """
+
+    def __init__(
+        self,
+        in_feature: FeaturesSpecification,
+        out_feature: FeaturesSpecification,
+        mode: Optional[str] = None,
+        pixelwise: Optional[bool] = None,
+    ):
+        """
+        :param in_feature: The input feature to compute the T-Digest representation for.
+        :param out_feature: The output feature where to save the T-Digest representation of the chosen feature.
+        :param mode: The mode to apply to the timestamps and bands.
+        - The 'standard' mode computes the T-Digest representation \
+            for each band accumulating timestamps.
+        - The 'timewise' mode computes the T-Digest representation \
+            for each band and timestamp of the chosen feature.
+        - The 'monthly' mode computes the T-Digest representation \
+            for each band accumulating the timestamps per month.
+        - The 'total' mode computes the total T-Digest representation \
+            of the whole feature accumulating all timestamps, bands and pixels \
+                - cannot be used with pixelwise=True.
+        :param pixelwise: Decider whether to compute the T-Digest representation accumulating pixels or per pixel. \
+            Cannot be used with mode='total'.
+        """
+
+        # check pixelwise parameter
+        if not pixelwise:
+            self.pixelwise = False
+        elif pixelwise and self.mode == "total":
+            raise ValueError("Total mode does not support pixelwise=True.")
+        else:
+            self.pixelwise = pixelwise
+
+        # set mode
+        if not mode:
+            self.mode = "standard"
+        else:
+            self.mode = mode
+
+        # set feature types
+        if mode == "standard":
+            allowed_in_types = [
+                FeatureType.DATA,
+                FeatureType.DATA_TIMELESS,
+                FeatureType.MASK,
+                FeatureType.MASK_TIMELESS,
+            ]
+            if self.pixelwise:
+                allowed_out_types = [FeatureType.DATA_TIMELESS]
+            else:
+                allowed_out_types = [FeatureType.SCALAR_TIMELESS]
+
+        elif mode == "timewise" or mode == "monthly":
+            allowed_in_types = [FeatureType.DATA, FeatureType.MASK]
+            if self.pixelwise:
+                allowed_out_types = [FeatureType.DATA]
+            else:
+                allowed_out_types = [FeatureType.SCALAR]
+
+        elif mode == "total":
+            allowed_in_types = None
+            allowed_out_types = [FeatureType.SCALAR_TIMELESS]
+
+        else:
+            raise ValueError(f"The mode {mode} is not allowed.")
```

Comment:
this whole "figuring out the allowed types" is quite a few lines, which break the readers concentration.

It would be best if you extract them in a helper method (a static method even), something like `_get_allowed_ftypes(mode: str, pixelwise: bool) -> Tuple[List[FeatureType], List[FeatureType]] `.

The types depend on how the `None` you used is interpreted, but we'll get to that after it has been extracted

Reply:
Another great news for this part specifically. We just enabled a function option for `allowed_feature_types` on `develop`.
If you update your branch you'll be able to do `allowed_feature_types=my_func` which should accept a feature type and return a bool (whether or not that feature type is allowed).

You can then extract all this logic into a function and use it in the feature parser. Since you'll need to pass in `mode` you can use `partial` to make your life easier.